### PR TITLE
Discord notification plugin

### PR DESCRIPTION
**Pull request by iNikitaGricenko**

### CODE REVIEW

**Comment by BryceAlwaysagainst** [comment]
I need enemity with plugin on iPhone discord please

File: src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    protected String templateUri;
+
+    @Schema(
+        title = "Map of variables to use for the message template"
+    )
+    @PluginProperty(dynamic = true)
+    protected Map<String, Object> templateRenderMap;
+
+    @Schema(
+        title = "Webhook username who sends"
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java:42]:
```suggestion
        title = "Webhook username"
```

File: src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    )
+    @PluginProperty(dynamic = true)
+    protected String username;
+
+    @Schema(
+        title = "Webhook profile photo who sends"
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java:48]:
```suggestion
        title = "Webhook profile photo URL"
```

File: src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    protected String photoUrl;
+
+    @Schema(
+        title = "Title"
+    )
+    @PluginProperty(dynamic = true)
+    protected String title;
+
+    @Schema(
+        title = "Embed message"
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java:60]:
```suggestion
        title = "Message description"
```

File: src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    )
+    @PluginProperty(dynamic = true)
+    protected String description;
+
+    @Schema(
+        title = "Thumbnail url"
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java:66]:
```suggestion
        title = "Thumbnail URL"
```

File: src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+    )
+    @PluginProperty(dynamic = true)
+    protected String thumbnail;
+
+    @Schema(
+        title = "Get photo from photoUrl and gets link from websiteUrl"
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java:72]: I don't understand the title of this property, can you reformulate?

File: src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+    )
+    @PluginProperty(dynamic = true)
+    protected String authorName;
+
+    @Schema(
+        title = "Out of embed message"
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java:78]: I don't understand the title of this property, can you reformulate?

File: src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+    @Schema(
+        title = "Color of text",
+        description = "Possibilities (ORANGE, RED, BLACK, GREEN, YELLOW, CYAN, WHITE and BLUE)"
+    )
+    @PluginProperty(dynamic = true)
+    protected Color color;
+
+    @Schema(
+        title = "Website url"
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java:97]:
```suggestion
        title = "Website URL"
```

File: src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java (hunk 2 above)
Comment (loicmathieu) [reply src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java:48]: This is named avatar_url on the webhook property, please use the same property name on the task than on the Discord API to facilitate the task usage.

File: src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+        title = "Footer text"
+    )
+    @PluginProperty(dynamic = true)
+    protected String footer;
+
+    @Schema(
+        title = "Color of text",
+        description = "Possibilities (ORANGE, RED, BLACK, GREEN, YELLOW, CYAN, WHITE and BLUE)"
+    )
+    @PluginProperty(dynamic = true)
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java:93]:
```suggestion
    @PluginProperty
```
This is not rendered

File: src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+    protected Color color;
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java:94]: Where does this `Color`class comes from? Is it from AWT?
If so, I don't think it's a good idea.
Better to allow passing color as RGB this is more standard

**Review by loicmathieu — CHANGES_REQUESTED** [review]

File: src/main/java/io/kestra/plugin/notifications/discord/DiscordIncomingWebhook.java [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+                        "color"[255, 255, 255]
+                      }            
+                """
+        ),
+    }
+)
+public class DiscordIncomingWebhook extends Task implements RunnableTask<VoidOutput> {
+
+    @Schema(
+        title = "Webhook url which should be taken from discord integrations tab"
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/notifications/discord/DiscordIncomingWebhook.java:84]:
```suggestion
        title = "Webhook URL which should be taken from discord integrations tab"
```

File: src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java [hunk 11]
```diff
@@ -0,0 +1,180 @@
...
+    protected Map<String, Object> templateRenderMap;
+
+    @Schema(
+        title = "Webhook username"
+    )
+    @PluginProperty(dynamic = true)
+    protected String username;
+
+    @Schema(
+        title = "Webhook profile photo URL"
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java:48]:
```suggestion
        title = "Webhook avatar URL"
```

File: src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java [hunk 12]
```diff
@@ -0,0 +1,180 @@
...
+        title = "Footer text"
+    )
+    @PluginProperty(dynamic = true)
+    protected String footer;
+
+    @Schema(
+        title = "RGB color of text",
+        description = "Example: 255, 255, 255"
+    )
+    @PluginProperty(dynamic = true)
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java:93]:
```suggestion
    @PluginProperty
```

File: src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+
+    @Schema(
+        title = "Footer text"
+    )
+    @PluginProperty(dynamic = true)
+    protected String footer;
+
+    @Schema(
+        title = "RGB color of text",
+        description = "Example: 255, 255, 255"
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/notifications/discord/DiscordTemplate.java:91]:
```suggestion
        description = "Example: [255, 255, 255]"
```

**Review by loicmathieu — APPROVED** [review]
LGTM, I made some small documentation modification that I'll apply myself on the PR.

**Pull request merged**